idaes/models_extra/gas_solid_contactors/unit_models/fixed_bed_1D.py

```diff
@@ -0,0 +1,2047 @@
+            blk.gas_solid_htc_eqn.activate()
+
+            # Activate gas phase energy balance equations
+            blk.gas_phase_heat_transfer.activate()
+            blk.gas_phase.enthalpy_balances.activate()
+
+            # Activate solid phase energy balance equations
+            blk.solid_phase_heat_transfer.activate()
+            blk.solid_enthalpy_balances.activate()
+
+            init_log.info("Initialize Energy Balances")
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                results = opt.solve(blk, tee=slc.tee, symbolic_solver_labels=True)
+            if check_optimal_termination(results):
+                init_log.info_high(
+                    "Initialization Step 4 {}.".format(idaeslog.condition(results))
+                )
+            else:
+                raise InitializationError(
+                    f"{blk.name} failed to initialize successfully. Please "
+                    f"check the output logs for more information."
+                )
+
+        # Initialize energy balance
+        if blk.config.energy_balance_type == EnergyBalanceType.none:
+            for t in blk.flowsheet().time:
+                for x in blk.length_domain:
+                    if (
+                        t != blk.flowsheet().config.time.first()
+                        and x != blk.length_domain.first()
+                    ):
+                        # Unfix gas temperature variables except at the inlet
+                        blk.gas_phase.properties[t, x].temperature.unfix()
+                        blk.gas_phase.properties[t, x].temperature = value(
+                            blk.solid_properties[0, x].temperature
+                        )
+
+            for t in blk.flowsheet().time:
+                if t != blk.flowsheet().config.time.first():
+                    for x in blk.length_domain:
+                        # Unfix solid temperature variables except initial
+                        blk.solid_properties[t, x].temperature = value(
+                            blk.solid_properties[0, x].temperature
+                        )
+                        blk.solid_properties[t, x].temperature.unfix()
+
+            blk.isothermal_gas_phase.activate()
+            blk.isothermal_solid_phase.activate()
+
+            init_log.info("Initialize Energy Balances")
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                results = opt.solve(blk, tee=slc.tee)
+            if check_optimal_termination(results):
+                init_log.info_high(
+                    "Initialization Step 4 {}.".format(idaeslog.condition(results))
+                )
+            else:
+                raise InitializationError(
+                    f"{blk.name} failed to initialize successfully. Please "
+                    f"check the output logs for more information."
+                )
+
+    def block_triangularization_initialize(
```

Comment:
This method needs to be documented so users know it exists.

As a side note for the future, I am working on a proposal that will integrate this directly into the initialize method as an option for all models.

Reply:
Sounds good. I'll include it in the documentation of the model.